components/frontend/src/widgets/Button.js

```diff
@@ -32,13 +61,17 @@ ActionButton.propTypes = {
     position: string,
 }
 
-export function AddDropdownButton({ itemSubtypes, itemType, onClick, allItemSubtypes }) {
+export function AddDropdownButton({ itemSubtypes, itemType, onClick, allItemSubtypes, usedItemSubtypeKeys }) {
     const [selectedItem, setSelectedItem] = useState(0) // Index of selected item in the dropdown
     const [query, setQuery] = useState("") // Search query to filter item subtypes
     const [menuOpen, setMenuOpen] = useState(false) // Is the menu open?
     const [popupTriggered, setPopupTriggered] = useState(false) // Is the popup triggered by hover or focus?
     const [showAllItems, setShowAllItems] = useState(false) // Show all itemSubTypes or only supported itemSubTypes?
-    const items = showAllItems ? allItemSubtypes : itemSubtypes
+    const [hideUsedItems, setHideUsedItems] = useState(false) // Hide itemSubTypes already used?
+    let items = showAllItems ? allItemSubtypes : itemSubtypes
+    if (hideUsedItems) {
```

Comment:
The naming of `showAllItems` is a bit confusing when we still take out some items afterwards.
Perhaps negate it to something like `onlyShowSupportedItems`?

Reply:
Renamed `showAllItems` to `showUnsupportedItems`.